main/webapp/modules/core/scripts/dialogs/add-rows-dialog.js

```diff
@@ -0,0 +1,101 @@
+const AddRowsDialog = (function (path) {
+  let level;
+  const $dialog = $(DOM.loadHTML("core", path));
+
+  class Row {
+    #isFlagged;
+    #isStarred;
+    #cells;
```

Comment:
I wonder if this class is necessary because since we are only supporting adding blank rows, we should be able to only send the number of blank rows (and not their JSON serialization) to the backend, no?

Reply:
Yes, we could do that, but I think in the future we would wish that designed this feature from the beginning to send rows that are blank, as opposed to the number of blank rows, see   https://github.com/OpenRefine/OpenRefine/pull/6461#discussion_r1576775598.